internal/destination/writer.go

```diff
@@ -74,36 +70,18 @@ func NewWriter(params writerParams) (*Writer, error) {
 		subject:     params.subject,
 		publisher:   jetstream,
 		publishOpts: params.getPublishOptions(),
-		canWrite:    atomic.Bool{},
 	}
 
-	w.startWrites()
-
 	return w, nil
 }
 
 // Write synchronously writes a record.
-func (w *Writer) Write(record sdk.Record) error {
-	// not redundant with Destination.Write
-	// writes can become unavailable when processing multiple records
-	if !w.canWrite.Load() {
-		return errWriteUnavailable
-	}
-
-	_, err := w.publisher.Publish(w.subject, record.Payload.After.Bytes(), w.publishOpts...)
+func (w *Writer) write(ctx context.Context, record sdk.Record) error {
+	publishOpts := append(w.publishOpts, nats.Context(ctx))
+	_, err := w.publisher.Publish(w.subject, record.Payload.After.Bytes(), publishOpts...)
```

Comment:
I just noticed one last thing - we should publish `record.Bytes()` to nats, then the user can choose the [output format](https://conduit.io/docs/connectors/output-formats) (this functionality is provided by the SDK via a middleware).